docs/writing-rules/pattern-syntax.md

```diff
@@ -316,65 +329,15 @@ baz.qux()
 
 ## Constants
 
-Languages supporting constants allow for constant propagation. In other words, the constant’s value is considered equivalent to a literal value. The following patterns will catch each respective code snippet:
-
-Go:
-
-The pattern `crypto.hash("MD5", ...)` matches:
-
-```go
-crypto.hash("MD5", "text")
-
-const algorithm = "MD5"
-crypto.hash(algorithm, "text")
-
-```
-
-Java:
+Semgrep performs basic constant propagation.
```

Comment:
This functionality can now be assumed true for all GA languages, this wasn't the case when this documentation was originally written.